### PR TITLE
[capi-media-vision][ACR-1701] Add ROI tracker CAPI for MediaVision

### PR DESCRIPTION
**Pull request by hancepark**
### Change Description ###

Describe your changes here.
New API set 'ROI Tracker' is added.

### API Changes ###
 - https://code.sec.samsung.net/jira/browse/ACR-1701

### CODE REVIEW

**Comment by omar-srbd** [comment]
@hancepark, hello.
Please add the file `docs/application/native/guides/multimedia/roi-tracker.md` to the `toc_all.md` file.
Thank you.

**Comment by hancepark** [comment]
Dear @omar-srbd 
Thank you, i add one more patch for your suggestion.

**Review by omar-srbd — COMMENTED** [review]
@hancepark, Hello.
I have reviewed the changed parts of the files on this PR and made some suggestions.
Please incorporate them at your earliest convenience.

**Comment by hancepark** [comment]
Dear @omar-srbd ,
thank you for suggestion, i applied all your things. Thank you

File: docs/application/native/guides/multimedia/roi-tracker.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# ROI Tracker
+
+ROI Tracker is one of the main features of the Media Vision API. This API allows users to obtain the proper ROI coordinates that users want to track in an image.
+For example, when an image and ROI coordinates are provided as input to this API, the Media Vision framework will process the given image and ROI coordinates from the decoded image data and will provide ROI coordinates within the given image.
```
Comment (omar-srbd) [inline docs/application/native/guides/multimedia/roi-tracker.md:4]:
```suggestion
ROI Tracker is one of the main features of the Media Vision API. This API allows users to obtain the proper ROI coordinates that users want to track in an image. For example, when an image and ROI coordinates are provided as input to this API, the Media Vision framework will process the given image and ROI coordinates from the decoded image data and will provide ROI coordinates within the given image.
```

File: docs/application/native/guides/multimedia/roi-tracker.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    int error_code = 0;
+
+    error_code = mv_create_source(&mv_source);
+    if (error_code != MEDIA_VISION_ERROR_NONE)
+        dlog_print(DLOG_ERROR, LOG_TAG, "error code = %d", error_code);
+
+    error_code = mv_create_engine_config(&imagedata.g_engine_config);
+    if (error_code != MEDIA_VISION_ERROR_NONE)
+        dlog_print(DLOG_ERROR, LOG_TAG, "error code= %d", error_code);
+    ```
```
Comment (omar-srbd) [inline docs/application/native/guides/multimedia/roi-tracker.md:44]: Global:
Change has been suggested to maintain consistency.
```suggestion
    ```

```

File: docs/application/native/guides/multimedia/roi-tracker.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+3. To Track the `sample.jpg` image, create a `handle` media vision ROI Tracker handle:
+
+    ```c
+    error_code = mv_roi_tracker_create(&handle);
+    if (error_code != MEDIA_VISION_ERROR_NONE)
+        dlog_print(DLOG_ERROR, LOG_TAG, "error code = %d", error_code);
+    ```
+
+4. Set tracker mode for running the tracker. The default mode is `MV_ROI_TRACKER_TYPE_BALANCE`:
```
Comment (omar-srbd) [inline docs/application/native/guides/multimedia/roi-tracker.md:67]:
```suggestion
4. Set tracker mode for running the tracker. The default mode is `MV_ROI_TRACKER_TYPE_BALANCE`:

```

File: docs/application/native/guides/multimedia/roi-tracker.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    ```c
+    int error_code = 0;
+    error_code = mv_engine_config_set_int_attribute(engine_cfg, MV_ROI_TRACKER_TYPE, (int) MV_ROI_TRACKER_TYPE_BALANCE);
+	if (ret != MEDIA_VISION_ERROR_NONE)
+		dlog_print(DLOG_ERROR, LOG_TAG, "error code= %d", error_code);
+    ```
```
Comment (omar-srbd) [inline docs/application/native/guides/multimedia/roi-tracker.md:73]:
```suggestion
    ```

```

File: docs/application/native/guides/multimedia/roi-tracker.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    For more information on the configuration attributes such as `MV_ROI_TRACKER_TYPE_BALANCE`, see Media Vision ROI Tracker API (in mobile and wearable applications).
```
Comment (omar-srbd) [inline docs/application/native/guides/multimedia/roi-tracker.md:74]:
```suggestion
    For more information on the configuration attributes, such as `MV_ROI_TRACKER_TYPE_BALANCE`, see Media Vision ROI Tracker API (in mobile and wearable applications).
```

File: docs/application/native/guides/multimedia/roi-tracker.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+5. Use `mv_roi_tracker_configure()` to configure `handle` ROI Tracker handle with `config`:
```
Comment (omar-srbd) [inline docs/application/native/guides/multimedia/roi-tracker.md:76]:
```suggestion
5. Use `mv_roi_tracker_configure()` to configure `handle` ROI Tracker handle with `config`:

```

File: docs/application/native/guides/multimedia/roi-tracker.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    ```c
+    error_code = ret = mv_roi_tracker_configure(handle, config);
+    if (error_code != MEDIA_VISION_ERROR_NONE)
+        dlog_print(DLOG_ERROR, LOG_TAG, "error code = %d", error_code);
+    ```
+
+6. Use `mv_roi_tracker_prepare()` to prepare ROI tracking. This uses the coordinates x, y, width, and height, that will be used for ROI tracking within the image:
```
Comment (omar-srbd) [inline docs/application/native/guides/multimedia/roi-tracker.md:83]:
```suggestion
6. Use `mv_roi_tracker_prepare()` to prepare ROI tracking. This uses the coordinates x, y, width, and height, that will be used for ROI tracking within the image:

```

File: docs/application/toc_all.md [hunk 8]
```diff
@@ -149,6 +149,7 @@
 ##### [Face Detection, Recognition, and Tracking](/application/dotnet/guides/multimedia/face-detection.md)
 ##### [Image Recognition and Tracking](/application/dotnet/guides/multimedia/image-recognition.md)
 ##### [Pose Detection](/application/dotnet/guides/multimedia/pose-detection.md)
+##### [ROI Tracker](/application/dotnet/guides/multimedia/roi-tracker.md)
```
Comment (omar-srbd) [inline docs/application/toc_all.md:152]:
```suggestion
##### [ROI Tracker](/application/native/guides/multimedia/roi-tracker.md)
```

**Review by omar-srbd — COMMENTED** [review]
@hancepark, thanks for implementing the changes as per my suggestion.
However, during another check, I found a few more issues that need to be addressed. Apart from these, there are a few suggestions that you missed from the last review, listing them below:

https://github.com/Samsung/tizen-docs/pull/1719#discussion_r1005393380
https://github.com/Samsung/tizen-docs/pull/1719#discussion_r1005394893
https://github.com/Samsung/tizen-docs/pull/1719#discussion_r1005395231
https://github.com/Samsung/tizen-docs/pull/1719#discussion_r1005401264

There is a mention of `mobile` and `wearable` in lines 10 & 74. Shouldn't they be hyperlinked which leads to their respective API pages?

File: docs/application/native/guides/multimedia/roi-tracker.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    ```
+
+4. Set tracker mode for running tracker. Defaut mode is `MV_ROI_TRACKER_TYPE_BALANCE`:
+    ```c
+    int error_code = 0;
+    error_code = mv_engine_config_set_int_attribute(engine_cfg, MV_ROI_TRACKER_TYPE, (int) MV_ROI_TRACKER_TYPE_BALANCE);
+	if (ret != MEDIA_VISION_ERROR_NONE)
+		dlog_print(DLOG_ERROR, LOG_TAG, "error code= %d", error_code);
+    ```
+    For more information on the configuration attributes such as `MV_ROI_TRACKER_TYPE_BALANCE`, see Media Vision ROI Tracker API (in mobile and wearable applications).
```
[the remark this answers is not in the document]
Comment (hancepark) [reply docs/application/native/guides/multimedia/roi-tracker.md:74]: There is no page for ROI Tracker yet.
I found example that other page has like below for following your suggestion. 

file: docs/application/native/guides/multimedia/face-detection.md

`To use the functions and data types of the Media Vision Face API (in [mobile](../../api/mobile/latest/group__CAPI__MEDIA__VISION__FACE__MODULE.html) and [wearable](../../api/wearable/latest/group__CAPI__MEDIA__VISION__FACE__MODULE.html) applications), include the `<mv_face.h>` header file in your application.)`

i understood  that i shoud make html page for hyperlinking mobile or wearble. But when i checked html page, it looks like auto-generated. 

Could you please let me know how to make HTML page?

[the remark this answers is not in the document]
Comment (omar-srbd) [reply docs/application/native/guides/multimedia/roi-tracker.md:74]: Hello @hancepark,
After a brief check on the API reference page for **mobile** and **wearable** of the Native profile, I have found links that may be associated with ROI tracker. I am attaching the links below for your reference:


**Wearable API:** https://docs.tizen.org/application/native/api/wearable/latest/group__CAPI__MEDIA__VISION__ROI__TRACKER__MODULE.html
**Mobile API:** https://docs.tizen.org/application/native/api/mobile/latest/group__CAPI__MEDIA__VISION__ROI__TRACKER__MODULE.html

Could you please check and update the files accordingly if the links are associated with the ROI Tracker feature.
Thank you.

**Comment by hancepark** [comment]
Dear @omar-srbd ,
Hello, i updated following your suggestion. 
And there's no respective API pages for ROI tracker. 
It seems that respective API pages is auto generated.
Or, Please let me know guide for this.
Thank you for your great help.

File: docs/application/native/guides/multimedia/roi-tracker.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (hancepark) [reply docs/application/native/guides/multimedia/roi-tracker.md]: Hello, Mr. @omar-srbd I added hyperlink:)
Thank you for your great help.

**Review by omar-srbd — COMMENTED** [review]
@hancepark, thanks for making the appropriate change to hyperlink the words **mobile** and **wearable**. However, there is another instance of **mobile** & **wearable** in **_line 76_** that needs to be hyperlinked. Please hyperlink them before we can proceed to the next step.

Thank you.

**Comment by hancepark** [comment]
Dear @omar-srbd ,
Hello, I'm sorry for disturbing you.
I fixed it.
Thank you

**Review by omar-srbd — APPROVED** [review]
@hancepark, thank you for incorporating all the suggested changes and taking my hyperlinks suggestion into consideration.

@asm-foysal @mijin85cho I am approving this PR for merging.

**Comment by hancepark** [comment]
Please merge this patch@asm-foysal @mijin85cho

**Comment by mijin85cho** [comment]
> Reference in new issue
> Edit
> Hide
> Delete

Hello, @asm-foysal Would you please check the approved PRs?
Also, please check if this is the last left PR to tizen_7.0_prepare.

As the wrap up, 
it would be better to merge tizen_7.0_prepare to master again to take the left things.
After that, tizen_7.0_prepare should have nothing left, so please don't accept additional PRs to this branch if any raised.

**Comment by asm-foysal** [comment]
Hi Catherine! 
I think this is the last PR for tizen 7.0 prepare. 
I'll review and merge it ASAP. 
Thanks for the note on not accepting PRs in this branch anymore. I'll redirect any PR to master if required.

**Pull request merged**